src/views/textview/TextSegment.js

```diff
@@ -162,7 +175,9 @@ export function TextSegment({
       return ChineseSegment(inputData, words);
     }
     if (lang === LANGUAGE_CODES.SANSKRIT || lang === LANGUAGE_CODES.PALI) {
-      return PaliSanskritSegment(inputData, words);
+      return SanskritSegment(inputData, words);
+    } else if (lang === LANGUAGE_CODES.PALI) {
```

Comment:
This does not make sense. Here you first ask if something is either sanskrit or pali, then when it is pali you get it to run the sanskritsegment code.

Reply:
This is a mistake, indeed.